### PR TITLE
SSCS-4095 Add bulk scanning happy path test

### PR DESCRIPTION
**Pull request by Rob-Doherty**
### WHAT?

These changes add all the Cucumber and Protractor functionality to perform an end-to-end journey for an appeal where all mandatory fields are present.

### WHY?

This is the first step towards a set of case worker tests that will catch any unintentional changes to the expected functionality of CCD configuration. 
 
### HOW?

The tests are written in Cucumber for easy readability, using Protractor to do the interactions with the browser, and Puppeteer to provide the headless Chrome support.
- The bulk scanning happy path test lives in `/e2e/features/bulk-scanning/create-appeal.feature`
- It uses the steps in the `/e2e/features/step-definitions/*.steps.ts` files to setup some initial test data
- Test data is found in `/e2e/features/data/scanned-case.js`
- The generated case is then moved into the "Case created" state

### CODE REVIEW

File: Jenkinsfile_nightly [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #!groovy
 
 properties([
-        // TODO: modify the chron trigger to a time period you want to run them
-        pipelineTriggers([cron('H 02 * * *')]),
+        pipelineTriggers([cron('H 22 * * *')]),
```
Comment (jackmaloney) [inline Jenkinsfile_nightly]: Does this mean the end to end tests only run overnight? Not when there might be a breaking change?

File: e2e/features/data/scanned-case.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  formData : [
+    { question: 'is_hearing_type_oral', answer: 'true' },
+    { question: 'is_hearing_type_paper', answer: 'false' },
+    { question: 'person1_first_name', answer: 'cde' },
+    { question: 'person1_last_name', answer: 'def' },
+    { question: 'person1_address_line1', answer: 'efg' },
+    { question: 'person1_address_line3', answer: 'fgh' },
+    { question: 'person1_address_line4', answer: 'ghi' },
+    { question: 'person1_postcode', answer: 'hij' },
+    { question: 'person1_nino', answer: 'ijk' }
```
Comment (jackmaloney) [inline e2e/features/data/scanned-case.js]: I think mrn_date has now been added as a mandatory field

File: e2e/features/idam-sign-in.feature [hunk 3]
```diff
@@ -1,10 +1,5 @@
 Feature: User authentication
 
-  Scenario: Should show Sign In form instead of Case List when user has *not* signed in
```
Comment (jackmaloney) [inline e2e/features/idam-sign-in.feature]: What is this feature and is it relevant for Bulk Scanning?

File: e2e/features/bulk-scanning/create-appeal.feature [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Feature: Create an bulk-scanned appeal
+
+  Scenario: Should end up in "appeal created" state when all mandatory fields are present
+    Given I am signed in as a Case Officer
+    And I have a bulk-scanned document with all mandatory fields
+    When I choose the next step "Create new case from exception"
+    Then the case should be in "Case created" state
```
Comment (jackmaloney) [inline e2e/features/bulk-scanning/create-appeal.feature]: Is Case created the right name for the state? Is it not 'Appeal Created'?

File: Jenkinsfile_nightly (hunk 1 above)
Comment (Rob-Doherty) [reply Jenkinsfile_nightly]: yes, unfortunately at the moment there is no connection between the upload of the CCD definition and the triggering of these tests, so the tests will always run nightly (as a safety net) and then will have to be manually triggered on the env where the changed definition is being tested (e.g. AAT, Demo, etc). (the manual triggering is already in `master`)

File: e2e/features/data/scanned-case.js (hunk 2 above)
Comment (Rob-Doherty) [reply e2e/features/data/scanned-case.js]: brilliant, I'll add that, thanks

File: e2e/features/bulk-scanning/create-appeal.feature (hunk 4 above)
Comment (Rob-Doherty) [reply e2e/features/bulk-scanning/create-appeal.feature]: thats an interesting question, I might have an old version of the definition

File: e2e/features/idam-sign-in.feature (hunk 3 above)
Comment (Rob-Doherty) [reply e2e/features/idam-sign-in.feature]: removed 👍

**Comment by Rob-Doherty** [comment]
@jackmaloney can you please re-review, I've made the changes based on our conversation

File: Jenkinsfile_nightly (hunk 1 above)
Comment (jackmaloney) [reply Jenkinsfile_nightly]: Ok sounds good

File: e2e/features/bulk-scanning/create-appeal.feature (hunk 4 above)
Comment (jackmaloney) [reply e2e/features/bulk-scanning/create-appeal.feature]: Case Created is the Bulk Scan definition name. We need to also be testing for 'Appeal Created' in SSCS CCD. Will find out today how we can link the 2 cases so you can easily find it for this test

File: e2e/pages/idam-sign-in.page.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { browser, $ } from 'protractor';
```
Comment (jackmaloney) [inline e2e/pages/idam-sign-in.page.ts]: Do you need this file if you have deleted the IDAM test?

**Comment by Rob-Doherty** [comment]
Thanks @jackmaloney, I've implemented your change regarding checking the "Appeal Created" state and pushed up - would you mind re-reviewing? thanks

**Pull request merged**